### ISSUE TITLE
Vrecord goes into crazy loop if it doesn't have configuration file

### ISSUE THREAD

**Issue by jfarbowitz** [amiaopensource/vrecord#26]
I deleted the config file and here is the garbage I saw when I ran vrecord:

No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: line 15: /Users/archivesx/.vrecord.conf: No such file or directory
No config file, restarting in config mode.
/usr/local/bin/vrecord: fork: Resource temporarily unavailable
/usr/local/bin/vrecord: fork: Resource temporarily unavailable
/usr/local/bin/vrecord: fork: Resource temporarily unavailable
/usr/local/bin/vrecord: fork: Resource temporarily unavailable

**Comment by jfarbowitz** [comment]
Then eventually it does this: 

Edting of preferences was canceled by the user.
Which VIDEO input are you using?
1) Composite
2) SDI
3) Component
4) S-Video

**Comment by dericed** [comment]
Ya it loops here: https://github.com/amiaopensource/vrecord/blob/master/vrecord#L15. With the current logic, it tries to load the config file and it is doesn't exist it tries to restart vrecord which leads to the same issue. I think line 15 and https://github.com/amiaopensource/vrecord/blob/master/vrecord#L27-L31 need to be merged together.

**Comment by jfarbowitz** [comment]
I merged them like so:

```
if [ -f "${config_file}" ] ; then
    . "${config_file}"
else
    echo -e "No configuration file, restarting in edit mode."
    touch "${config_file}"
    $(basename "${0}") -e
fi
```

The result is that if you don't have a config file, even if you ran `$ vrecord -p` you still get sent to edit mode. Is that ok?

**Comment by jfarbowitz** [comment]
Should be fixed in 3ca0448.
